Added support for decoding .debug_pubtypes and .debug_pubnames sections

rvijayc opened the pull request:
This pull request is meant to fix #201 . It adds support for parsing .debug_pubtypes and .debug_pubnames sections from an ELF file. 

There is an example file (examples/dwarf_pubnames_types.py) added that exercises the new features.

eliben commented on elftools/dwarf/namelut.py:
```diff
@@ -0,0 +1,188 @@
+from ..construct import CString, Struct
+
+NameLUTEntry = collections.namedtuple('NameLUTEntry',
+        'cu_ofs die_ofs')
```
Nit: don't wrap this

rvijayc replied:
I have resolved this in the latest commit.

eliben commented on examples/dwarf_pubnames_types.py:
```diff
@@ -0,0 +1,111 @@
+        
+            # dump all entries in .debug_pubnames section.
+            print('Dumping .debug_pubnames table ...')
+            print('-'*66)
```
Here and everywhere else: spaces around operators 

```
'-' * 66
```

rvijayc replied:
I have resolved this in the latest commit.

rvijayc commented:
That's a good idea. I'll add the readelf support when I get a chance (along with other review comments), and add a new commit.

rvijayc commented:
I've added the following in the latest commit:
- Added readelf tests for debug_pubnames and debug_pubtypes.
- fixed all review comments.
- Formatted doc strings.
